spec/invoice_item_spec.rb

```diff
@@ -0,0 +1,145 @@
+require 'rspec'
+require './lib/invoice_item'
+
+describe InvoiceItem do
+  describe '#initialize' do
+    it 'exists' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item).is_a? InvoiceItem
+    end
+
+    it 'returns the integer id' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.id).to eq 6
+      expect(invoice_item.id).is_a? Integer
+    end
+
+    it 'returns the item id' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
```

Comment:
I still don't understand this. I would like to talk more about this because the value 10.99 being turned into 0.001099 makes no sense in terms of memory usage.

Reply:
@redferret I think this storage has more to do with consistency since prices *could* get really large, like 100000000000 dollars or whatever, which would take a lot of storage. Anyway, it's less a choice than a requirement here, I think, since the spec harness will tell us how this has to be formatted.